src/adapters/openai.ts

```diff
@@ -19,14 +66,31 @@ export const requestParser: AgenticModel.RequestParser<OpenAi.AiModel> = (
 ) => {
   const request: AiAdapter.Input<OpenAi.AiModel> = {
     messages: messages.map((m) => {
+      const role = StateRoleToOpenAiRole[m.role];
       return {
-        role: m.role,
+        ...(m.stop_reason ? { finish_reason: StateStopReasonToOpenAiStopReason[m.stop_reason] } : {}),
+        role,
         content: m.content,
+        // NOTE: this is very ugly, we need to better handle different shape of messages
```

Comment:
@jpwilliams I wouldn't say I like this current state; too much error-prone IMO

We could extract this in a helper that transforms based on the `role`